### PR TITLE
Update Sensitive Data page with instructions for before-send (SDK)

### PR DESCRIPTION
**Pull request by MimiDumpling**
Sensitive Data page was missing a code snippet and additional instructions regarding `before-send`. Updated to include them.

image

### CODE REVIEW

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 1]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+    });
+    ```
+    
+- Attaching user/email information (if it is considered SD or not permitted)
+
+    Don't do:
+    
+    ```javascript
+    Sentry.configureScope((scope) => {
+      scope.setUser({username: user.email});
```
Comment (lobsterkatie) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```suggestion
      scope.setUser({email: user.email});
```

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 2]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+    ```javascript
+    console.log("user's name is: " + user.name);
+    ```
+    
+    Instead:
+    
+    ```
+    // 1. don't log SD/PII
+    // 2. use `beforeBreadcrumb` to filter it out from breadcrumbs before it is attached
+    // 3. Disable logging breadcrumb integration (e.g. <https://docs.sentry.io/platforms/javascript/?platform=browser#alternative-way-of-setting-an-integration>)
```
Comment (lobsterkatie) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```suggestion
    // 3. Disable logging breadcrumb integration (e.g. <https://docs.sentry.io/platforms/javascript/?platform=browser#breadcrumbs-1>)
```

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 3]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
 
 ## Custom Event Processing in the SDK
 
-In the SDKs you can configure a `before-send` function which is invoked before an event is sent and can be used to modify the event data and remove sensitive data. See [_Filtering Events_]({%- link _documentation/error-reporting/configuration/filtering.md -%}) for more information.
+In the SDKs, you can configure a `before-send` function which is invoked before an event is sent and can be used to modify the event data and remove sensitive data. Using `before-send` in the SDKs to **scrub any data before it is attached** as a payload request to [sentry.io](http://sentry.io/) is the recommended scrubbing approach, as SD is not attached to the event and therefore never makes it to Sentry.
```
Comment (lobsterkatie) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```suggestion
In the SDKs, you can configure a `before-send` function, which is invoked before an event is sent and can be used to modify the event data and remove sensitive information. Using `before-send` in the SDKs to **scrub any data before it is sent** is the recommended scrubbing approach, as SD never leaves the local environment.
```

**Comment by lobsterkatie** [comment]
A few small suggestions, but otherwise LGTM!

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 4]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+### Examples
+
+- Attaching PII/SD
+
+    Don't do:
+    
+    ```javascript
+    Sentry.configureScope((scope) => {
+      scope.setTag('birthday', '08/03/1990');
+    });
```
Comment (HazAT) [inline src/collections/_documentation/data-management/sensitive-data.md]: We can use the simple variant here:
```
Sentry.setTag('birthday', '08/03/1990');
```

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 5]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+    ```
+    
+    Instead, anonymize (by using `checksum`, `hash`, etc.):
+    
+    ```javascript
+    Sentry.configureScope((scope) => {
+      scope.setTag('birthday', checksum_or_hash('08/12/1990'));
+    });
```
Comment (HazAT) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```
Sentry.setTag('birthday', checksum_or_hash('08/12/1990'));
```

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 6]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+    ```
+    
+- Attaching user/email information (if it is considered SD or not permitted)
+
+    Don't do:
+    
+    ```javascript
+    Sentry.configureScope((scope) => {
+      scope.setUser({username: user.email});
+    });
```
Comment (HazAT) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```
Sentry.setUser({email: user.email});
````

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 7]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+    ```    
+
+    Instead, anonymize (use `id`, `username`, etc.):
+    
+    ```javascript
+    Sentry.configureScope((scope) => {
+      scope.setUser({id: user.id});
+    });
```
Comment (HazAT) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```
Sentry.setUser({id: user.id});
```

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 8]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+    
+    // or
+    
+    Sentry.configureScope((scope) => {
+      scope.setUser({username: user.username});
+    });
```
Comment (HazAT) [inline src/collections/_documentation/data-management/sensitive-data.md]:
```
Sentry.setUser({username: user.username});
```

**Comment by MimiDumpling** [comment]
@HazAT applied the changes. Thanks!

File: src/collections/_documentation/data-management/sensitive-data.md [hunk 9]
```diff
@@ -19,7 +19,89 @@ Turning this option on is required for certain features in Sentry to work, but a
...
+      scope.setUser({email: user.email});
+    });
+    ```    
+
+    Instead, anonymize (use `id`, `username`, etc.):
+    
+    ```javascript
+    Sentry.configureScope((scope) => {
+      Sentry.setUser({id: user.id});
+    });
```
Comment (HazAT) [inline src/collections/_documentation/data-management/sensitive-data.md]: 😬 
You can completely remove the `configureScope` everywhere
```
Sentry.configureScope((scope) => {
...
});
```
and just keep the inner part, like so:
```
Sentry.setUser({id: user.id});
```

Comment (MimiDumpling) [reply src/collections/_documentation/data-management/sensitive-data.md]: Oh, I see! Okie. Will make changes today.

**Comment by MimiDumpling** [comment]
@HazAT 😸 I removed `configureScope` from the code snippets

**Pull request merged**